blockchain/blockchain.go

```diff
@@ -2262,15 +2264,13 @@ func (bc *BlockChain) InsertHeaderChain(chain []*types.Header, checkFreq int) (i
 	}
 
 	// Make sure only one thread manipulates the chain at once
-	bc.chainmu.Lock()
-	defer bc.chainmu.Unlock()
+	bc.mu.Lock()
+	defer bc.mu.Unlock()
 
 	bc.wg.Add(1)
 	defer bc.wg.Done()
 
 	whFunc := func(header *types.Header) error {
-		bc.mu.Lock()
```

Comment:
@hqjang-pepper I'm not sure we can remove this. This function is called in InsertrHeaderChain below.

If we remove here, InsertHeaderChain will be executed without holding mutex.


Could you please run this modification with klaytn-deploy with heavy load?

Reply:
Got it. I'll conduct load test with this code.